File: PYTHON/bubble_sort.py
from tkinter import FALSE


def bubble_sort(arr):
    sorted = FALSE
    unsortedIndex = len(arr) - 1
    while not sorted:
        sorted = True
        for i in range(unsortedIndex):
            sorted = True
            if(arr[i+1] < arr[i]):
                next = arr[i+1]
                arr[i+1] = arr[i]
                arr[i]= next
                sorted = FALSE
                i +=1
        unsortedIndex -=1
    return arr
    

print(bubble_sort([5,4,3,2,1]))

File: JAVASCRIPT/bubble_sort.js
function bubble_sort(arr){
    let swapped = true;
    let lastSortedIndex = 0;
    while(swapped){
        swapped = false;
      for(let i = 0; i < arr.length - lastSortedIndex; i++){
        console.log(i + 1 + "--pass")
        if(arr[i+1] < arr[i]){
            let next = arr[i+1];
            arr[i+1] = arr[i];
            arr[i] = next;
            swapped = true;
        }
      }
      lastSortedIndex += 1;
    }
    return arr;
}

console.log(bubble_sort([5,4,3,2,1]))
// console.log(bubble_sort([2,1,5,2,34,5,6,7,8,9]))